### DISCUSSION TITLE
Does `db push` not create a `_prisma_migrations` table?

### DISCUSSION THREAD

**Discussion by asfaltboy** [prisma/prisma#20523] [Q&A] [answered]
### Question

I'm trying to only run migrations when necessary in my tests, and am using `prisma db push` command, but it seems it does not create the table `_prisma_migrations` in my db. Note that I'm using a schema for parallel tests. Here's the full command:

```shell
DB_URL='postgresql://user:pass@localhost:5432/my_app_test?schema=test-40200' yarn prisma db push
yarn run v1.22.19
$ .../my-app/node_modules/.bin/prisma db push
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": PostgreSQL database "my_app_test", schema "test-40200" at "localhost:5432"

🚀  Your database is now in sync with your Prisma schema. Done in 128ms

✔ Generated Prisma Client (5.1.0 | library) to ./node_modules/@prisma/client in 82ms

✨  Done in 0.89s.
```

However, when running `yarn prisma migrate reset` the table `_prisma_migrations` is created fine.

Is this the intended behavior? And if so, should we update the documentation to mention this fact?

For context: I've been wanting to speed up our Prisma integration tests, by using transaction rollback and skipping running migrations between tests. I'm following an article that explains how to do just that here: https://selimb.hashnode.dev/speedy-prisma-pg-tests , the guide proposes we detect if migrations have run by inspecting the `_prisma_migrations` table, but this seems to not work. (any more?

### Information about Prisma Schema, Client Queries and Environment (optional)

- OS: macOS arm64
- Database: PostgreSQL
- Node.js version: v18.17.0

- Run `prisma -v` to see your Prisma version and paste it
```
prisma                  : 5.1.0
@prisma/client          : 5.1.0
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine a9b7003df90aa623086e4d6f4e43c72468e6339b (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Schema Engine           : schema-engine-cli a9b7003df90aa623086e4d6f4e43c72468e6339b (at node_modules/@prisma/engines/schema-engine-darwin-arm64)
Schema Wasm             : @prisma/prisma-schema-wasm 5.1.0-28.a9b7003df90aa623086e4d6f4e43c72468e6339b
Default Engines Hash    : a9b7003df90aa623086e4d6f4e43c72468e6339b
Studio                  : 0.492.0
```

**Comment by ludralph** [comment, marked as answer]
Hi @asfaltboy 👋 

When you use the `prisma db push` command, it applies the changes from your Prisma schema to the database directly without creating the `_prisma_migrations` table. This is because `prisma db push` is designed for rapid prototyping and development, and it assumes that you are using a disposable database that you can easily reset if needed. See this section of the [documentation](https://www.prisma.io/docs/concepts/components/prisma-migrate/db-push)

On the other hand, the `prisma migrate reset` command is intended for development environments. See the [documentation](https://www.prisma.io/docs/concepts/components/prisma-migrate/migrate-development-production#reset-the-development-database)

If you want to use the `_prisma_migrations` table to track the migration state in your tests, you should use the `prisma migrate` commands instead of `prisma db push`. Specifically, you should use `prisma migrate deploy` to generate and apply migrations, which will create and update the _prisma_migrations table as expected. See the [documentation](https://www.prisma.io/docs/concepts/components/prisma-migrate/migrate-development-production#production-and-testing-environments)

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by asfaltboy** [reply]
Thanks I did browse those docs, but missed the fact that it's not creating the table. Should we add a small note for future selves?

**Reply by ludralph** [reply]
@asfaltboy there is a note section in the [documentation](https://www.prisma.io/docs/concepts/components/prisma-migrate/db-push#:~:text=Notes%3A,on%20your%20situation.) that highlights this.

**Reply by ludralph** [reply]
Hi @asfaltboy 

Do you have any more questions around this issue? If no, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!

**Reply by asfaltboy** [reply]
As a newbie, this sentence did not connect:

> The migrations table will not be updated, and no migration files will be generated.

I've created [a tiny docs PR](https://github.com/prisma/docs/pull/5130) with to change this to say:

> The migrations table `_prisma_migrations` will not be created or updated, and no migration files will be generated.

P.s: thanks for your help @ludralph

**Reply by ludralph** [reply]
Thank you for creating the PR 🙌 . A team member will take a look.
